src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj

```diff
@@ -5,14 +5,21 @@
     <DebugOptimization Condition="'$(Configuration)' == 'Release'">OPT</DebugOptimization>
   </PropertyGroup>
   <PropertyGroup>
+    <CoreCompileDependsOn>$(CoreCompileDependsOn);GenerateVersionFile</CoreCompileDependsOn>
     <DocumentationFile>$(MSBuildThisFileDirectory)System.Runtime.CompilerServices.Unsafe.xml</DocumentationFile>
     <ProjectGuid>{04BA3E3C-6979-4792-B19E-C797AD607F42}</ProjectGuid>
-    <IncludePath>include\$(TargetFramework)</IncludePath>
-    <IncludePath Condition="$(TargetFramework.StartsWith('net4'))">include\netfx</IncludePath>
+    <IncludePath>$([MSBuild]::NormalizeDirectory('$(BaseIntermediateOutputPath)', '$(TargetFramework)-$(TargetFrameworkSuffix)-$(Configuration)', 'version'))</IncludePath>
+    <IncludePath Condition="'$(TargetFrameworkSuffix)' == ''">$([MSBuild]::NormalizeDirectory('$(BaseIntermediateOutputPath)', '$(TargetFramework)-$(Configuration)', 'version'))</IncludePath>
+    <CoreAssembly>System.Runtime</CoreAssembly>
+    <ExtraMacros Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)' or '$(TargetFramework)' == 'netcoreapp2.0'">#define netcoreapp</ExtraMacros>
     <IlasmFlags>$(IlasmFlags) -INCLUDE=$(IncludePath) -DEBUG=$(DebugOptimization)</IlasmFlags>
     <TargetFrameworks>$(NetCoreAppCurrent);netstandard2.0;netcoreapp2.0;netstandard1.0;net45</TargetFrameworks>
     <ExcludeCurrentNetCoreAppFromPackage>true</ExcludeCurrentNetCoreAppFromPackage>
   </PropertyGroup>
+  <PropertyGroup Condition="$(TargetFramework.StartsWith('net4'))">
+    <CoreAssembly>mscorlib</CoreAssembly>
+    <ExtraMacros>#define net45</ExtraMacros>
```

Comment:
You could avoid this hardcoding if you just set all DefineConstants (and make sure we run after the AddImplicitDefineConstants target.

eg:
```xml
    <ItemGroup>
     <_defineConstants Include="$(DefineConstants.Split(';,'))" />
    </ItemGroup>
...
      <_VersionFileContents>
<![CDATA[
#define CORE_ASSEMBLY "$(CoreAssembly)"
#define ASSEMBLY_VERSION "$(_AssemblyVersion)"
#define CORE_ASSEMBLY_VERSION "$(_CoreAssemblyVersion)"
@(_defineConstants->'#define %(Identity)', '
')
// Metadata version: v4.0.30319
.assembly extern CORE_ASSEMBLY
{
  .publickeytoken = ($(_coreAssemblyPublicKeyToken) )
  .ver CORE_ASSEMBLY_VERSION
}
 ]]> 
      </_VersionFileContents>
```

Reply:
Same for previous use of ExtraMacros as well.

Reply:
@ericstj we need to add some parsing code to the Microsoft.Net.Sdk.IL to make this work for all the tfms. 
These implicit constants are set based on TFI & TFV and also the language Version. I have a working thing(I will push the pr) I Think we can merge this PR without this.